lib/config.module.ts

```diff
@@ -30,24 +30,25 @@ export class ConfigModule {
    * @param options
    */
   static forRoot(options: ConfigModuleOptions = {}): DynamicModule {
+    let config = { ...process.env };
```

Comment:
In this case, Joi will validate all the process variables, even these like `SHELL` or `NODE` and this doesn't make too much sense. What if someone set `allowUnknown` to `false`? In this case, the validation will **always** throw an error. 

Generally, the `validationSchema` option was intended to be used with `dotenv` and `.env` files. Maybe, if we want to allow validating process environment variables too, we should add a flag for this (another option that would force set `allowUnknown` to `false` as well and assign `process.env` to `config` as above).